assets/src/dashboard/components/typeaheadOptions/index.js

```diff
@@ -118,19 +126,63 @@ const TypeaheadOptions = ({ isOpen, items, maxItemsVisible = 5, onSelect }) => {
   }, [hoveredIndex, items, onSelect, isOpen]);
 
   useEffect(() => {
-    if (listRef.current) {
-      listRef.current.scrollToItem(0);
+    if (!listRef.current) {
+      return () => {};
     }
-    setHoveredIndex(0);
-  }, [items]);
+
+    listRef.current.children[0].focus();
+
+    return () => {};
+  }, []);
+
+  // we only want to set an existing value for currentSelection when the dropdown is newly opened
+  useEffect(() => {
+    if (isOptionMenuAlreadyOpen) {
+      return () => {};
+    }
+
+    setIsOptionMenuAlreadyOpen(true);
+
+    const selectionToCheckFor =
+      currentSelection && currentSelection.toLowerCase().trim();
+    const existingValueOnMenuOpen = selectionToCheckFor
+      ? items.findIndex(
+          (item) =>
+            (item.value &&
+              item.value.toLowerCase() === selectionToCheckFor.toLowerCase()) ||
+            item.label.toLowerCase() === selectionToCheckFor.toLowerCase()
```

Comment:
nit: You don't need to call `.toLowerCase()` on `selectionToCheckFor` again since it will have always been lowercased on line 147

Reply:
true! thanks, i missed that when i moved the value to selectionToCheckFor.